From 5a8dc928863daf72eec91c45d5785687e1132333 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:29 -0400
Subject: [block] export blk_finish_request

Message-id: <1402670238-13916-42-git-send-email-snitzer@redhat.com>
Patchwork-id: 83881
O-Subject: [RHEL7.1 PATCH 041/150] block: export blk_finish_request
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 12120077b2612a243d158605640cd39266906667
Author: Christoph Hellwig <hch@lst.de>
Date:   Wed Apr 16 09:44:59 2014 +0200

    block: export blk_finish_request

    This allows to mirror the blk-mq code flow for more a more readable I/O
    completion handler in SCSI.

    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-core.c b/block/blk-core.c
index 6775d0d..8e051be 100644
--- a/block/blk-core.c
+++ b/block/blk-core.c
@@ -2497,7 +2497,7 @@ EXPORT_SYMBOL_GPL(blk_unprep_request);
 /*
  * queue lock must be held
  */
-static void blk_finish_request(struct request *req, int error)
+void blk_finish_request(struct request *req, int error)
 {
  if (blk_rq_tagged(req))
   blk_queue_end_tag(req->q, req);
@@ -2523,6 +2523,7 @@ static void blk_finish_request(struct request *req, int error)
   __blk_put_request(req->q, req);
  }
 }
+EXPORT_SYMBOL(blk_finish_request);
 
 /**
  * blk_end_bidi_request - Complete a bidi request
diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 9931256..118991b 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -954,6 +954,7 @@ extern struct request *blk_fetch_request(struct request_queue *q);
  */
 extern bool blk_update_request(struct request *rq, int error,
           unsigned int nr_bytes);
+extern void blk_finish_request(struct request *rq, int error);
 extern bool blk_end_request(struct request *rq, int error,
        unsigned int nr_bytes);
 extern void blk_end_request_all(struct request *rq, int error);
-- 
1.7.1